holoviews/ipython/display_hooks.py

```diff
@@ -59,7 +60,13 @@ def render(obj, **kwargs):
     # Drop back to png if pdf selected, notebook PDF rendering is buggy
     if renderer.fig == 'pdf':
         renderer = renderer.instance(fig='png')
-    return renderer.html(obj, **kwargs)
+
+    data, metadata = renderer.components(obj, **kwargs)
```

Comment:
``renderer.components`` makes sense to me though I can imagine it could be useful for giving the components in situations outside of jupyterlab/notebook. Is the output dependent on the renderer ``mode``? Is the plan to make ``components`` useful in different contexts or for different environments?

Reply:
It's not specific to the notebook, e.g. in theory you could use the output in a flask app, and with a ``SocketIOCommManager`` it would all work.

Reply:
Ok. I was just making sure the intention behind the method was to be as general as possible.

Reply:
Sort of unfortunate you decided to reuse the IPython comms in labmode as it would have made for a good test. May have a go at a Flask+SocketIO example at some point.